src/test/java/io/dropwizard/grpc/server/GrpcServerTests.java

```diff
@@ -0,0 +1,242 @@
+
+                    assertEquals((double) shutdownPeriodInMillis, shutdownDurationInMillis, DELTA_IN_MILLIS);
+                } catch (final Exception e) {
+                    throw new RuntimeException(e);
+                }
+            });
+    }
+
+    @Test
+    public void shouldThrowWhenShutdownTimeoutIsExceeded() throws Exception {
+        final long shutdownPeriodInMillis = 1000L;
+        final long slowRequestDurationInMillis = 3000L;
+
+        testShutdownPeriodHonoured(shutdownPeriodInMillis, slowRequestDurationInMillis,
+            (requestDurationInMillisFut, shutdownDurationInMillisFut) -> {
+                try {
+                    try {
+                        requestDurationInMillisFut.get(shutdownPeriodInMillis + DELTA_IN_MILLIS, TimeUnit.SECONDS);
+                        fail("Request should have thrown an exception");
+                    } catch (final ExecutionException e) {
+                        assertEquals(StatusRuntimeException.class, e.getCause().getClass());
+                        assertEquals(Code.CANCELLED, ((StatusRuntimeException) e.getCause()).getStatus().getCode());
+                    }
+                } catch (final Exception e) {
+                    throw new RuntimeException(e);
+                }
+            });
+    }
+
+    @Test
+    public void grpcServerGetsStopped() {
+        final DropwizardTestSupport<TestConfiguration> testSupport =
+                new DropwizardTestSupport<>(TestApplication.class, resourceFilePath("grpc-test-config.yaml"));
+
+        ManagedChannel channel = null;
+        try {
+            testSupport.before();
+            channel = createPlaintextChannel(testSupport);
+            final PersonServiceGrpc.PersonServiceBlockingStub client = PersonServiceGrpc.newBlockingStub(channel);
+
+            testSupport.after();
+
+            try {
+                // this should fail as the server is now stopped
+                client.getPerson(GetPersonRequest.newBuilder().setName("blah").build());
+                fail("Request should have failed.");
+            } catch (final Exception e) {
+                assertEquals(StatusRuntimeException.class, e.getClass());
+                assertEquals(Code.UNAVAILABLE, ((StatusRuntimeException) e).getStatus().getCode());
+            }
+        } finally {
+            testSupport.after();
+            shutdownChannel(channel);
+        }
+    }
+
+    /**
+     * Concurrently sends a long-running request and shuts down the server, then executes the assertions. It guarantees
+     * that the shutdown is requested while the server is processing the long-running request.
+     *
+     * @param shutdownPeriodInMillis the time the gRPC server is supposed to wait for shutdown
+     * @param slowRequestDurationInMillis the minimum time the long-running request takes
+     */
+    private void testShutdownPeriodHonoured(final long shutdownPeriodInMillis, final long slowRequestDurationInMillis,
```

Comment:
Nice!

The variable names get a bit long. Can you replace long with dropwizard's Duration so this becomes a bit more readable?